From 853c67d218e03db1998632d23dd1a07676e9c142 Mon Sep 17 00:00:00 2001
From: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Date: Thu, 1 Aug 2024 16:20:20 +0530
Subject: [PATCH] QSDK: mac80211: Fix warn on if mesh metric is zero

During performence measurement, stats is disabled through boost
performence script. So, host is not receiving the metric values
from firmware and the value becomes zero during mpath finding.
This issue addressed through static mpath. But,in RvR far range
test, connection becomes unstable and getting this warnon
continuously. So, add a check to prevent this warnon if the
stats is disabled.

Patch-dependency:
	913-001-wifi-mac80211-avoid-tx-pending-queue-in-native-wifi-.patch

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
---
 net/mac80211/mesh_hwmp.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/net/mac80211/mesh_hwmp.c
+++ b/net/mac80211/mesh_hwmp.c
@@ -375,8 +375,10 @@ u32 airtime_link_metric_get(struct ieee8
 			return MAX_METRIC;
 
 		rate = ewma_mesh_tx_rate_avg_read(&sta->mesh->tx_rate_avg);
-		if (WARN_ON(!rate))
+		if (!rate) {
+			WARN_ON(!local->hw.perf_mode);
 			return MAX_METRIC;
+		}
 
 		err = (fail_avg << ARITH_SHIFT) / 100;
 	}
